src/java.desktop/windows/classes/com/sun/java/swing/plaf/windows/WindowsIconFactory.java

```diff
@@ -878,12 +877,47 @@ public void paintIcon(Component c, Graphics g, int x, int y) {
                             if (icon == null) {
                                 skin = xp.getSkin(c, part);
                                 skin.paintSkin(g, x + OFFSET, y + OFFSET, state);
+                            } else {
+                                skin = xp.getSkin(c, part);
```

Comment:
`skin = xp.getSkin(c, part);` can be taken out of if-else block.

Reply:
ok